application/Core/CLI/Commands/Record/RecordCommand.cs

```diff
@@ -35,10 +37,18 @@ public int Execute(RecordOptions options)
                 OutputFormatter.PrintDebug(startRecordingMessage);
                 sessionManager.StartRecording();
 
+                // We start our interactive commandline so the user
+                // can still interact with the application.
+                var commandLine = new InteractiveCommandLine();
+                commandLine.Launch();
+
                 // Run message loop required for Windows hooks
                 NativeMethods.DoWin32MessageLoop();
```

Comment:
It's odd to start the message loop here, but stop it inside the `Start()` function of `InteractiveCommandLine`.

Reply:
If we start the message loop in the Start() method of the InteractiveCommandLine our implementation is broken. There are two major reasons for that:

1. The message loop is not spawned on the main thread which will cause it not to work correctly at all, as this is an requirement for a message loop.

2. If we start the message loop in the Start method we block the thread in which the method runs and so for user interactive cancel is no longer possible.

As calling the message loop is a necessary requirement only for CollectingModules while recording I would keep it here.